src/controllers/confirmation.controller.ts

```diff
@@ -51,25 +51,31 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       });
   }
 
-  const token = req.chSession.accessToken() as string;
-  let apiResponseData: any;
-  try {
-    // TODO  LFA-1406 Add isSubmitted flag to prevent this being sent twice
+  const isSubmitted: boolean = getPromiseToFileSessionValue(req.chSession, ALREADY_SUBMITTED);
+  if (!isSubmitted) {
+    const token = req.chSession.accessToken() as string;
+    let apiResponseData: any;
+    try {
 
-    const axiosResponse: AxiosResponse = await callPromiseToFileAPI(companyProfile.companyNumber,
-        token, isStillRequired);
+        const axiosResponse: AxiosResponse = await callPromiseToFileAPI(companyProfile.companyNumber,
+            token, isStillRequired);
 
-    apiResponseData = axiosResponse.data;
+        apiResponseData = axiosResponse.data;
+        await updatePromiseToFileSessionValue(req.chSession, NEW_DEADLINE, apiResponseData.filing_due_on);
+        await updatePromiseToFileSessionValue(req.chSession, ALREADY_SUBMITTED, true);
 
-    logger.debug(`Response data returned from the PTF api call : ${JSON.stringify(apiResponseData)}`);
-  } catch (e) {
-    logger.error("Error processing application " + JSON.stringify(e));
-    return next(e);
+        logger.debug(`Response data returned from the PTF api call : ${JSON.stringify(apiResponseData)}`);
+    } catch (e) {
+        logger.error("Error processing application " + JSON.stringify(e));
+        await updatePromiseToFileSessionValue(req.chSession, ALREADY_SUBMITTED, false);
+        return next(e);
+    }
+  } else {
+      logger.error("Form already submitted, not processing again");
   }
 
   if (isStillRequired) {
-    const filingDueOn = apiResponseData.filing_due_on;
-
+    const filingDueOn =  getPromiseToFileSessionValue(req.chSession, NEW_DEADLINE);
```

Comment:
Two spaces to right of equals sign. Should/can npm lint be picking up these formatting issues?

Reply:
Removed extra space.